cpp/src/parquet/file_writer.cc

```diff
@@ -140,37 +140,35 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
 
     const int32_t column_ordinal = next_column_index_++;
     const auto& path = col_meta->descr()->path();
+    const ColumnProperties& column_properties = properties_->column_properties(path);
     auto meta_encryptor =
         file_encryptor_ ? file_encryptor_->GetColumnMetaEncryptor(path->ToDotString())
                         : nullptr;
     auto data_encryptor =
         file_encryptor_ ? file_encryptor_->GetColumnDataEncryptor(path->ToDotString())
                         : nullptr;
-    auto ci_builder = page_index_builder_ && properties_->page_index_enabled(path) &&
-                              properties_->statistics_enabled(path)
+    auto ci_builder = page_index_builder_ && column_properties.page_index_enabled() &&
+                              column_properties.statistics_enabled()
                           ? page_index_builder_->GetColumnIndexBuilder(column_ordinal)
                           : nullptr;
-    auto oi_builder = page_index_builder_ && properties_->page_index_enabled(path)
+    auto oi_builder = page_index_builder_ && column_properties.page_index_enabled()
                           ? page_index_builder_->GetOffsetIndexBuilder(column_ordinal)
                           : nullptr;
-    auto codec_options = properties_->codec_options(path)
-                             ? properties_->codec_options(path).get()
-                             : nullptr;
+    const CodecOptions* codec_options = column_properties.codec_options()
```

Comment:
Should we use explicit type here? Can we use `auto` here?

```suggestion
    const auto* codec_options = column_properties.codec_options()
```

Reply:
I prefer explicit type since we know the type here and it's unlikely to be changed. But we can wait for others comment here

Reply:
(It's not a strong opinion.)